### ISSUE TITLE
[Linter] Proposal: `prefer_wildcard_variables` lint rule

### ISSUE THREAD

**Issue by ricardodalarme** [dart-lang/sdk#59638]
### TLDR

Add a lint rule to prefer wildcard variables (`_`) over multiple underscores (`_, __, ___`) in unused parameters.

### Motivation

The upcoming Dart 3.7.0 release introduces support for wildcard variables (`_`) that can be reused for multiple unused parameters. Previously, developers commonly relied on distinct variables like `_, __, ___` to indicate unused parameters. This practice is now redundant with the introduction of `_`, which improves clarity and reduces visual noise in the codebase.

#### Reference

- https://github.com/dart-lang/sdk/blob/main/CHANGELOG.md#wildcard-variables

### Lint Rule Implementation

**Rule Name**: `prefer_wildcard_variables`  
**Description**: Warns when distinct variable names such as `__, ___, etc.` are used for unused parameters, recommending the use of `_` instead.  
**Severity Level**: WARNING (as it’s a stylistic improvement and non-breaking).  

#### Bad

```dart
ListView.builder(
  itemCount: 10,
  itemBuilder: (_, __) {
    return ListTile(
      title: Text('Item'),
    );
  },
);

```

#### Good

```dart
ListView.builder(
  itemCount: 10,
  itemBuilder: (_, _) {
    return ListTile(
      title: Text('Item'),
    );
  },
);
```

### Behavior

- **Triggers a warning**: When multiple distinct unused variable names (`__, ___`) are declared.  
- **Does not trigger**: For valid wildcard use cases, such as `(_, _, _)`.

### Benefits

1. **Improved Code Readability**: Reduces unnecessary differentiation in unused parameter names.
2. **Consistent Dart Style**: Encourages developers to adopt features introduced in newer Dart versions.
3. **Simpler Refactoring**: Avoids subtle issues where `_`, `__`, and `___` might accidentally be referenced or misinterpreted.

**Comment by srawlins** [comment]
CC @pq @kallentu sounds like a win to me!

**Comment by kallentu** [comment]
Ah yes, we've agreed upon making a lint that's exactly this. We may call it `unnecessary_underscores` or something more specific than `prefer_wildcard_variables` though.

For more context: https://github.com/dart-lang/sdk/issues/56595

**Comment by srawlins** [comment]
I like that name. `too_many_underscores`. `why_so_many_underscores`. 😁

**Comment by pq** [comment]
Hi! This has been implemented as `unnecessary_underscores` and should be in 3.7.

If it doesn't behave as expected, please raise issues.

Thanks!
